### PR TITLE
PEP8 code style for analyze.py 

### PR DESCRIPTION
**Pull request by ampresent**

### CODE REVIEW

**Comment by ampresent** [comment]
This PR is still under development. But I request for comments on the first 2 patches.

File: boundary/analyze.py [hunk 1]
```diff
@@ -238,29 +264,29 @@ def check_redirect_mangled(f, meta):
...
-            if file in config['sdcr_srcs']:
-                func_class['sdcr_fns'].add(signature)
-                decls[signature] = fn['decl_str']
-
-            if file in config['mod_hdrs']: hdr_sym['fn'].append(fn)
-            if init: func_class['init'].add(signature)
-            if publ: global_fn_dict[name] = file
+            fn = dotdict(fn)
+            fn.signature = tuple(fn.signature)
+            func_class.fn.add(fn.signature)
```
Comment (chenshanpei) [inline boundary/analyze.py:317]: Tab ? why it's different ?

File: boundary/analyze.py [hunk 2]
```diff
@@ -194,26 +193,53 @@ def check_redirect_mangled(f, meta):
...
             return True
         # If the caller is optimized too, check if it's unsafe recursively
-        if from_sym in func_class['mangled'] or from_sym not in func_class['in_vmlinux']:
+        if from_sym in func_class.mangled or from_sym not in func_class.in_vmlinux:
             if check_redirect_mangled(from_sym, meta):
                 return True
     return False
 
+def func_class_arithmetics(fns):
+    fns.callback -= fns.interface
```
Comment (chenshanpei) [inline boundary/analyze.py:247]: commit log:

border: use dotdict and underscope to shorten lines  --> there is no underscope now. (fns)

File: boundary/analyze.py [hunk 3]
```diff
@@ -343,17 +343,24 @@ class dotdict(dict):
...
     with open(tmpdir + 'boundary_extract.yaml', 'w') as f:
         dump(dict(config), f, Dumper)
+
+    tnt_fmt = 'TAINTED_FUNCTION({},{})'
+    und_fmt = '"{}", {}'
+    cb_fmt = "EXPORT_CALLBACK({fn}, {ret}, {params})"
+    export = "EXPORT_PLUGSCHED({fn}, {ret}, {params})"
+    mod_fmt = '__mod_{}'
+    unds, taints = [], []
+    for fn in func_class.undefined:
```
Comment (chenshanpei) [inline boundary/analyze.py]: Insert an emptly line.

File: boundary/analyze.py [hunk 4]
```diff
@@ -344,23 +344,22 @@ class dotdict(dict):
     with open(tmpdir + 'boundary_extract.yaml', 'w') as f:
         dump(dict(config), f, Dumper)
 
-    tnt_fmt = 'TAINTED_FUNCTION({},{})'
+    tnt_fmt = 'TAINTED_FUNCTION({},{})\n'
```
Comment (chenshanpei) [inline boundary/analyze.py:400]: A more formal commit log is required.

File: boundary/analyze.py [hunk 5]
```diff
@@ -39,29 +41,40 @@ def all_meta_files():
...
+    There are four pitfalls because of disagreement between vmlinux and
+    gcc-plugin, illustrated with examples below,
+
+      - Disagreement 1: vmlinux thinks XXX is in core.c
+                        plugsched thinks it's in kernel/sched/core.c
+      - Disagreement 2: vmlinux thinks XXX is in core.c
+                        plugsched thinks it's in sched.h
+      - Disagreement 3: vmlinux thinks XXX is in usercopy_64.c,
+                        plugsched thinks it's in core.c
+      - Disagrement: 4: vmlinux optimizes XXX->XXX.isra(or .constprop).1
```
Comment (chenshanpei) [inline boundary/analyze.py:63]: Disagrement ---> Disagreement

File: boundary/analyze.py [hunk 6]
```diff
@@ -93,8 +106,9 @@ def find_in_vmlinux(vmlinux_elf):
         file = filename
         # Disagreement 4
         if '.' in key:
-            # If function A has at least one mangled version, eg. A.isra, then function A
-            # may be called through the mangled one. But A.cold doesn't lead to this problem
+            # If function A has at least one mangled version, eg. A.isra
+            # Then function A may be called by the mangled one.
+            # But A.cold doesn't lead to this problem,
```
Comment (chenshanpei) [inline boundary/analyze.py:109]: Multi-line annotation, please use """ .

File: boundary/analyze.py [hunk 7]
```diff
@@ -125,49 +139,63 @@ def find_in_vmlinux(vmlinux_elf):
...
-            return to_sym
-    return None
-
 def inflect(initial_insiders, edges):
-    global __insiders
-    __insiders = copy.deepcopy(initial_insiders)
+    """
+    Mark functions called by outsiders as outsiders too, unless they're
+    interface or sidecar or callback functions.
+    """
```
Comment (chenshanpei) [inline boundary/analyze.py:147]: """Mark functions called by outsiders as outsiders too, unless they're
interface or sidecar or callback functions.
"""

File: boundary/analyze.py [hunk 8]
```diff
@@ -195,18 +227,22 @@ def check_redirect_mangled(f, meta):
...
+        # When caller is optimized too, check if it's unsafe recursively
         if from_sym in func_class.mangled or from_sym not in func_class.in_vmlinux:
             if check_redirect_mangled(from_sym, meta):
                 return True
     return False
 
 def func_class_arithmetics(fns):
+    """
+    Core algorithm of plugsched. This works by doing set operations.
+    """
```
Comment (chenshanpei) [inline boundary/analyze.py:246]: Single line annotation

File: boundary/analyze.py [hunk 9]
```diff
@@ -9,28 +9,20 @@
...
 
+chain = _chain.from_iterable
 # Use set as the default sequencer for yaml
 Loader.add_constructor(resolver.BaseResolver.DEFAULT_SEQUENCE_TAG,
-                       lambda loader, node: set(loader.construct_sequence(node)))
-Dumper.add_representer(set, lambda dumper, node: dumper.represent_list(node))
-Dumper.add_representer(str,
-                       lambda dumper, data: dumper.represent_scalar(u'tag:yaml.org,2002:str', data))
+                       lambda ldr, node: set(ldr.construct_sequence(node)))
+Dumper.add_representer(set, lambda dmpr, node: dmpr.represent_list(node))
```
Comment (chenshanpei) [inline boundary/analyze.py:17]: Please keep consistent with extact.py.  (loader / dumper). Don't go overboard with code brevity.

File: boundary/analyze.py [hunk 10]
```diff
@@ -251,31 +258,36 @@ def func_class_arithmetics(fns):
...
     """dot.notation access to dictionary attributes"""
     __getattr__ = dict.get
     __setattr__ = dict.__setitem__
     __delattr__ = dict.__delitem__
 
+
 if __name__ == '__main__':
     vmlinux = sys.argv[1]
+    # tmp directory to store middle files
     tmpdir = sys.argv[2]
```
Comment (chenshanpei) [inline boundary/analyze.py:280]: keep consistent with extract.py

tmpdir --> tmp_dir
modpath -->mod_path

File: boundary/analyze.py [hunk 11]
```diff
@@ -285,34 +297,38 @@ class dotdict(dict):
...
-        'init':      set(),
+        'fn': set(),
+        'init': set(),
         'interface': set(),
-        'callback':  set(),
-        'mod_fns':   set(),
-        'sdcr_fns':  set(),
+        'callback': set(),
+        'mod_fns': set(),
+        'sdcr_fns': set(),
```
Comment (chenshanpei) [inline boundary/analyze.py:305]: You can adjust the key order to make it neat.

File: boundary/analyze.py [hunk 12]
```diff
@@ -139,19 +139,18 @@ def get_in_any(fn, files):
...
+
     def inflect_one(edge):
         to_sym = tuple(edge['to'])
         if to_sym in insiders:
             from_sym = tuple(edge['from'])
-            if from_sym not in insiders and \
-            from_sym not in func_class.border and \
-            from_sym not in func_class.init and \
-            from_sym not in func_class.sidecar:
+            if from_sym not in insiders | func_class.inflect_cut:
```
Comment (chenshanpei) [inline boundary/analyze.py:153]: not in ( insiders | func_class.inflect_cut)

**Pull request merged**